pkg/csi/service/fcd/controller.go

```diff
@@ -72,15 +74,112 @@ func (c *controller) CreateVolume(
 	req *csi.CreateVolumeRequest) (
 	*csi.CreateVolumeResponse, error) {
 
-	return nil, nil
+	params := req.GetParameters()
+
+	// Volume Name
+	volName := req.GetName()
+	if len(volName) == 0 {
+		myUUID, err := uuid.NewUUID()
+		if err != nil {
+			log.Errorf("Failed to generate a UUID. Err: %v", err)
+			return nil, err
+		}
+		volName = myUUID.String()
+	}
+
+	// Volume Size - Default is 10 GiB
+	volSizeBytes := int64(DefaultGbDiskSize * GbInBytes)
+	if req.GetCapacityRange() != nil {
+		volSizeBytes = int64(req.GetCapacityRange().GetRequiredBytes())
+	}
+	volSizeMB := int64(volumeutil.RoundUpSize(volSizeBytes, GbInBytes)) * 1024
+
+	// Volume Type
+	datastoreType := vclib.TypeDatastoreCluster
+	volType := params[AttributeFirstClassDiskParentType]
```

Comment:
Any additional checking needed here for if this parameter is missing? looks like it is required.

Reply:
all of these fields must be provided, with update to check if each one is set and error out if it isnt